pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,373 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2014 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "jquery",
+    "base1/cockpit",
+    "kubernetes/client",
+    "base1/mustache"
+], function($, cockpit,kubernetes,Mustache) {
+    "use strict";
+
+    var appdeployer = {};
+    var client = kubernetes.k8client();
+
+    function debug() {
+        if (window.debugging == "all" || window.debugging == "deploy")
+            console.debug.apply(console, arguments);
+    }
+
+    function failure(ex) {
+        console.warn(ex);
+    }
+
+    function isJsonString(str) {
+        try {
+            JSON.parse(str);
+        } catch (e) {
+            return false;
+        }
+        return true;
+    }
+
+    function get_kentities(entity_key) {
+        var elist = [];
+        if (client[entity_key]) {
+            var el = client[entity_key];
+            for (var i = 0; i < el.length; i++) {
+                elist.push(el[i].metadata.name);
+            }
+        }
+        return elist;
+    }
```

Comment:
Is the goal of this function to copy the array of entities? If so, we should probably just use either javascript or jQuery to do that:

```
/* via javascript, copy an array: http://stackoverflow.com/questions/7486085/copying-array-by-value-in-javascript */
var elist = client.pods.slice();
```

```
/* via jQuery, copy an array: http://api.jquery.com/jQuery.merge/ */
var elist = $.merge([], client.pods);
```